zenoh-plugin-mqtt/src/lib.rs

```diff
@@ -211,38 +222,44 @@ async fn run(
             .expect("Unable to get the current runtime, which should not happen.");
-        if let Err(e) = rt.block_on(ntex::rt::System::new(MqttPlugin::DEFAULT_NAME).run_local(
-            async move {
-                let server = match tls_config {
-                    Some(tls) => ntex::server::Server::build().bind(
-                        "mqtt",
-                        config.port.clone(),
-                        move |_| {
-                            chain_factory(TlsAcceptor::new(tls.clone()))
-                                .map_err(|err| MqttError::Service(MqttPluginError::from(err)))
-                                .and_then(create_mqtt_server(
-                                    zsession.clone(),
-                                    config.clone(),
-                                    auth_dictionary.clone(),
-                                ))
-                        },
-                    )?,
-                    None => ntex::server::Server::build().bind(
-                        "mqtt",
-                        config.port.clone(),
-                        move |_| {
-                            create_mqtt_server(
+
+        let fut = ntex::rt::System::new(MqttPlugin::DEFAULT_NAME).run_local(async move {
+            let server = match tls_config {
+                Some(tls) => {
+                    ntex::server::Server::build().bind("mqtt", config.port.clone(), move |_| {
+                        chain_factory(TlsAcceptor::new(tls.clone()))
+                            .map_err(|err| MqttError::Service(MqttPluginError::from(err)))
+                            .and_then(create_mqtt_server(
                                 zsession.clone(),
                                 config.clone(),
                                 auth_dictionary.clone(),
-                            )
-                        },
-                    )?,
-                };
-                // Disable catching the signal inside the ntex, or we can't stop the plugin.
-                server.workers(1).disable_signals().run().await
-            },
-        )) {
+                            ))
+                    })?
+                }
+                None => {
+                    ntex::server::Server::build().bind("mqtt", config.port.clone(), move |_| {
+                        create_mqtt_server(
+                            zsession.clone(),
+                            config.clone(),
+                            auth_dictionary.clone(),
+                        )
+                    })?
+                }
+            };
+            // Disable catching the signal inside the ntex, or we can't stop the plugin.
+            tokio::select! {
+                _ = server.workers(1).disable_signals().run() => {
+                    println!("Server done");
+                    std::io::Result::Ok(())
+                }
+                _ = rx.recv_async() => {
+                    println!("Dropping all");
```

Comment:
Please remove this `println` or log it at `debug` level.

Reply:
changed to tracing::debug